src/nbl/video/CVulkanPhysicalDevice.cpp

```diff
@@ -397,25 +399,51 @@ std::unique_ptr<CVulkanPhysicalDevice> CVulkanPhysicalDevice::create(core::smart
 
         //vulkan12Properties.denormBehaviorIndependence;
         //vulkan12Properties.denormBehaviorIndependence;
-        properties.limits.shaderSignedZeroInfNanPreserveFloat32 = vulkan12Properties.shaderSignedZeroInfNanPreserveFloat32;
-        properties.limits.shaderDenormPreserveFloat32 = vulkan12Properties.shaderDenormPreserveFloat32;
-        properties.limits.shaderDenormFlushToZeroFloat32 = vulkan12Properties.shaderDenormFlushToZeroFloat32;
-        properties.limits.shaderRoundingModeRTEFloat32 = vulkan12Properties.shaderRoundingModeRTEFloat32;
-        properties.limits.shaderRoundingModeRTZFloat32 = vulkan12Properties.shaderRoundingModeRTZFloat32;
-        properties.limits.shaderSignedZeroInfNanPreserveFloat16 = vulkan12Properties.shaderSignedZeroInfNanPreserveFloat16;
-        properties.limits.shaderDenormPreserveFloat16 = vulkan12Properties.shaderDenormPreserveFloat16;
-        properties.limits.shaderDenormFlushToZeroFloat16 = vulkan12Properties.shaderDenormFlushToZeroFloat16;
-        properties.limits.shaderRoundingModeRTEFloat16 = vulkan12Properties.shaderRoundingModeRTEFloat16;
-        properties.limits.shaderRoundingModeRTZFloat16 = vulkan12Properties.shaderRoundingModeRTZFloat16;
-        properties.limits.shaderSignedZeroInfNanPreserveFloat64 = vulkan12Properties.shaderSignedZeroInfNanPreserveFloat64;
-        properties.limits.shaderDenormPreserveFloat64 = vulkan12Properties.shaderDenormPreserveFloat64;
-        properties.limits.shaderDenormFlushToZeroFloat64 = vulkan12Properties.shaderDenormFlushToZeroFloat64;
-        properties.limits.shaderRoundingModeRTEFloat64 = vulkan12Properties.shaderRoundingModeRTEFloat64;
-        properties.limits.shaderRoundingModeRTZFloat64 = vulkan12Properties.shaderRoundingModeRTZFloat64;
 
+        if (!vulkan12Properties.shaderSignedZeroInfNanPreserveFloat32)
+            return nullptr;
+        if (!vulkan12Properties.shaderDenormPreserveFloat32)
+            return nullptr;
+        if (!vulkan12Properties.shaderDenormFlushToZeroFloat32)
+            return nullptr;
+        if (!vulkan12Properties.shaderRoundingModeRTEFloat32)
+            return nullptr;
+        if (!vulkan12Properties.shaderRoundingModeRTZFloat32)
+            return nullptr;
+
+        if (properties.limits.shaderFloat16) {
+            if (!vulkan12Properties.shaderSignedZeroInfNanPreserveFloat16)
+                return nullptr;
+            if (!vulkan12Properties.shaderDenormPreserveFloat16)
+                return nullptr;
+            if (!vulkan12Properties.shaderDenormFlushToZeroFloat16)
+                return nullptr;
+            if (!vulkan12Properties.shaderRoundingModeRTEFloat16)
+                return nullptr;
+            if (!vulkan12Properties.shaderRoundingModeRTZFloat16)
+                return nullptr;
+        }
+
+        if (properties.limits.shaderFloat64) {
```

Comment:
`properties.limits.shaderFloat` are not initialized until line 925

Please be diligent in doing your work and follow my advice https://github.com/Devsh-Graphics-Programming/Nabla/pull/829#discussion_r1945494266

Reply:
Either you initialize your `properties.limits` first or you need to use `deviceProperties2.`